src/main/java/org/takes/facets/auth/codecs/CcAES.java

```diff
@@ -44,77 +43,56 @@
  * AES codec which supports 128 bits key.
  *
  * <p>The class is immutable and thread-safe.
- *
  * @author Jason Wong (super132j@yahoo.com)
  * @version $Id$
  * @since 0.13.8
  */
-@EqualsAndHashCode(of = { "origin", "secret", "enc", "dec" })
+@EqualsAndHashCode(of = {"origin", "key"})
 public final class CcAES implements Codec {
+    /**
+     * The block size constant.
+     */
+    private static final int BLOCK = 16;
 
     /**
-     * The cipher for encryption.
+     * Original codec.
      */
-    private final transient Cipher enc;
+    private final transient Codec origin;
 
     /**
-     * The cipher for decryption.
+     * The encryption key.
      */
-    private final transient Cipher dec;
+    private final transient byte[] key;
 
     /**
-     * Original codec.
+     * The algorithm parameter spec for cipher.
      */
-    private final transient Codec origin;
+    private transient AlgorithmParameterSpec spec;
 
     /**
-     * The AES secret key object.
+     * Tracks the spec variable is initialized or not.
      */
-    private final transient SecretKey secret;
+    private transient boolean initialized;
```

Comment:
@sebing this approach breaks rule http://www.yegor256.com/2014/06/09/objects-should-be-immutable.html. Please see what Yegor proposes instead:http://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html